data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/CommonTopicConfig.java

```diff
@@ -0,0 +1,224 @@
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import org.opensearch.dataprepper.model.types.ByteCount;
+
+import java.time.Duration;
+
+/**
+ * This class has topic configurations which are common to all Kafka plugins - source, buffer, and sink.
+ * <p>
+ * Be sure to only add to this configuration if the setting is applicable for all three types.
+ */
+public class CommonTopicConfig {
+    static final boolean DEFAULT_AUTO_COMMIT = false;
+    static final Duration DEFAULT_SESSION_TIMEOUT = Duration.ofSeconds(45);
+    static final String DEFAULT_AUTO_OFFSET_RESET = "earliest";
+    static final Duration DEFAULT_THREAD_WAITING_TIME = Duration.ofSeconds(5);
+    static final Duration DEFAULT_MAX_RECORD_FETCH_TIME = Duration.ofSeconds(4);
+    static final String DEFAULT_FETCH_MAX_BYTES = "50mb";
+    static final Integer DEFAULT_FETCH_MAX_WAIT = 500;
+    static final String DEFAULT_FETCH_MIN_BYTES = "1b";
+    static final Duration DEFAULT_RETRY_BACKOFF = Duration.ofSeconds(10);
+    static final Duration DEFAULT_RECONNECT_BACKOFF = Duration.ofSeconds(10);
+    static final String DEFAULT_MAX_PARTITION_FETCH_BYTES = "1mb";
+    static final Duration DEFAULT_MAX_POLL_INTERVAL = Duration.ofSeconds(300);
+    static final Integer DEFAULT_CONSUMER_MAX_POLL_RECORDS = 500;
+    static final Integer DEFAULT_NUM_OF_WORKERS = 2;
+    static final Duration DEFAULT_HEART_BEAT_INTERVAL_DURATION = Duration.ofSeconds(5);
+
+
+    @JsonProperty("name")
+    @NotNull
+    @Valid
+    private String name;
+
+    @JsonProperty("group_id")
+    @Valid
+    @Size(min = 1, max = 255, message = "size of group id should be between 1 and 255")
+    private String groupId;
+
+    @JsonProperty("workers")
+    @Valid
+    @Size(min = 1, max = 200, message = "Number of worker threads should lies between 1 and 200")
+    private Integer workers = DEFAULT_NUM_OF_WORKERS;
+
+    @JsonProperty("auto_commit")
+    private Boolean autoCommit = DEFAULT_AUTO_COMMIT;
+
+    @JsonProperty("session_timeout")
+    @Valid
+    @Size(min = 1)
+    private Duration sessionTimeOut = DEFAULT_SESSION_TIMEOUT;
+
+    @JsonProperty("auto_offset_reset")
+    private String autoOffsetReset = DEFAULT_AUTO_OFFSET_RESET;
+
+    @JsonProperty("thread_waiting_time")
+    private Duration threadWaitingTime = DEFAULT_THREAD_WAITING_TIME;
+
+    @JsonProperty("max_partition_fetch_bytes")
+    private String maxPartitionFetchBytes = DEFAULT_MAX_PARTITION_FETCH_BYTES;
+
+    @JsonProperty("fetch_max_bytes")
+    private String fetchMaxBytes = DEFAULT_FETCH_MAX_BYTES;
+
+    @JsonProperty("fetch_max_wait")
+    @Valid
+    @Size(min = 1)
+    private Integer fetchMaxWait = DEFAULT_FETCH_MAX_WAIT;
+
+    @JsonProperty("fetch_min_bytes")
+    private String fetchMinBytes = DEFAULT_FETCH_MIN_BYTES;
```

Comment:
I think a lot of these are not applicable to Producer. Things like `fetch_max_bytes`, `auto_commit` etc are consumer specific.